### DISCUSSION TITLE
FAQs

### DISCUSSION THREAD

**Discussion by ikanakova** [fast4x/RiMusic#370] [Q&A]
Can you think of any other questions for the [FAQs](https://github.com/fast4x/RiMusic/wiki/FAQs)?

**Comment by fast4x** [comment]
I’ve added more. If you have new ideas, enter.

**Comment by ValentinLeTallec** [comment]
Hi,
I don't know if it's a question for the FAQ, but I least it's one I'm wondering about : How do I use the dynamic theme ?
Setting it does not seem to change anything, and there are no options to change colors. I did see RiMusic themed while looking around in the UI discussion, but I could not figure out how.
(I'm on Android 8, so maybe it's a feature of more recent Android versions)

**Reply by fast4x** [reply]
Hi, Dynamic theme change background color in player and other parts, according to colors in the cover of song in now playing.

**Reply by ValentinLeTallec** [reply]
Oh, makes sense 😊
Thanks

**Comment by KyNarec** [comment]
What is the Visualizer for?

**Reply by fast4x** [reply]
Visualizer is a visual effect wich show music effect based on now playing song.
Enabled It from settings.

**Reply by KyNarec** [reply]
Ahhhh ok, Thanks 😊

**Comment by jingoniii** [comment]
Hello,
I am wondering where all of my downloaded music went?

**Reply by fast4x** [reply]
Hello, normally the downloaded music is under Data > Files while the cache in Data > Cache still depends on the system, it may also be different.

**Comment by aneesh1122** [comment]
How does RiMusic download songs? Does it use yt-dlp?

**Reply by fast4x** [reply]
it download from url

**Reply by aneesh1122** [reply]
> it download from url

You get the video URL from YouTube. How do you extract the audio?

**Reply by fast4x** [reply]
is player wich extract...

**Reply by OrangeZXZ** [reply]
> > it download from url
> 
> You get the video URL from YouTube. How do you extract the audio?

RiMusic uses something similar to youtube it just reads the video from the link (and then saves the informations from the cash)

Thats what i think

**Reply by aneesh1122** [reply]
> > > it download from url
> > 
> > You get the video URL from YouTube. How do you extract the audio?
> 
> RiMusic uses something similar to youtube it just reads the video from the link (and then saves the informations from the cash)
> 
> Thats what i think

Well, it doesn't download the video at all. It only extracts the audio that's why it downloads like 3-4 mb on average.

This app uses the YouTube API but I'm not sure how it downloads.

Maybe it just takes the streams and caches it in our mobile as 'downloads'.
